tools/cli/internal/changelog/outputfilter/squash.go

```diff
@@ -0,0 +1,211 @@
+		"response-required-property-added":            SquashResponseRequiredFieldAdded,
+		"response-required-property-removed":          SquashResponseRequiredFieldRemoved,
+		"response-optional-property-added":            SquashResponseOptionalFieldAdded,
+		"response-optional-property-removed":          SquashResponseOptionalFieldRemoved,
+		"response-property-became-required":           SquashResponseFieldBecameRequired,
+		"request-property-became-required":            SquashRequestFieldBecameRequired,
+		"new-required-request-property":               SquashRequestFieldAdded,
+		"request-property-removed":                    SquashRequestFieldRemoved,
+		"new-optional-request-property":               SquashNewOptionalRequestProperty,
+		"response-optional-property-became-read-only": SquashResponseOptionalFieldBecomeReadonly,
+	}
+}
+
+// EntryMappings groups entries by ID and then by OperationID
+func newEntriesMapPerIDAndOperationID(entries []*Entry) map[string]map[string][]*Entry {
+	result := make(map[string]map[string][]*Entry)
+
+	for _, entry := range entries {
+		code := entry.ID
+		operationID := entry.OperationID
+
+		// Ensure the code map exists
+		if _, exists := result[code]; !exists {
+			result[code] = make(map[string][]*Entry)
+		}
+
+		// Append the entry to the appropriate operationID slice
+		result[code][operationID] = append(result[code][operationID], entry)
+	}
+
+	return result
+}
+
+func squashEntries(entries []*Entry) ([]*Entry, error) {
+	entriesMap := newEntriesMapPerIDAndOperationID(entries)
+	squashHandlers := newSquashHandlers()
+	squashedEntries := []*Entry{}
+
+	for _, entry := range entries {
+		// if no squash handlers implemented for entry's code,
+		// just append the entry to the result
+		if _, ok := squashHandlers[entry.ID]; !ok {
+			squashedEntries = append(squashedEntries, entry)
+			continue
+		}
+	}
+
+	for id, handler := range squashHandlers {
+		entryMapPerOperationID, ok := entriesMap[id]
+		if !ok {
+			continue
+		}
+
+		entries, err := handler(entryMapPerOperationID)
+		if err != nil {
+			return nil, err
+		}
+
+		squashedEntries = append(squashedEntries, entries...)
+	}
+
+	return squashedEntries, nil
+}
+
+type squashedEntries struct {
+	valuesNotSquashed []string
+	valuesToSquash    map[string]struct{}
+}
+
+// squashEntriesByValues squashes entries based on the given parameters
+func squashEntriesByValues(
+	operation string,
+	entriesGroupedByOperationID map[string][]*Entry,
+	expectedNumberOfValues,
+	squashIdx int,
+	pluralizeFrom,
+	pluralizeTo string,
+) ([]*Entry, error) {
+	if len(entriesGroupedByOperationID) == 0 {
+		return nil, nil
+	}
+
+	result := []*Entry{}
+
+	for _, opEntries := range entriesGroupedByOperationID {
+		squashMap, err := newSquashMap(operation, opEntries, expectedNumberOfValues, squashIdx)
+		if err != nil {
+			return nil, err
+		}
+
+		templateEntry := opEntries[0]
+		templateText := templateEntry.Text
+
+		for _, squashData := range squashMap {
```

Comment:
this is the same logic we have in `squash_utils.go`. I debugged the code locally via the unit tests that I have written in field_test.go, and there are some part of this logic that are not used. I am going to migrate all the transformations and then see if we can refactor this logic to make it simpler